In [4]:
import pandas as pd
import numpy as np

# Read all movies
get the full movie list from https://bechdeltest.com/

In [38]:
df = pd.read_json('http://bechdeltest.com/api/v1/getAllMovies')

#### look how many film are present

In [39]:
df.tail()

id  rating                             title  year    imdbid
10131  10995       3  Winnie-the-Pooh: Blood and Honey  2023  19623240
10132  10996       3                   Haunted Mansion  2023   1695843
10133  11009       3                Portable Door, The  2023  11820950
10134  11010       3          Anatomie d&#39;une chute  2023  17009710
10135  11015       3                      Slotherhouse  2023  15422224

#### look at last film on Bechdel dataset from Kaggle

In [50]:
df_old = pd.read_csv("../data/Bechdel_kaggle.csv", index_col=0)

In [51]:
df_old

Unnamed: 0                       title    year  rating  dubious  \
0              0            Passage de Venus  1874.0     0.0      0.0   
1              1           La Rosace Magique  1877.0     0.0      0.0   
2              2  Sallie Gardner at a Gallop  1878.0     0.0      0.0   
3              3           Le singe musicien  1878.0     0.0      0.0   
4              4     Athlete Swinging a Pick  1881.0     0.0      0.0   
...          ...                         ...     ...     ...      ...   
9368        9368                   Love Hard  2021.0     2.0      0.0   
9369        9369                     Cruella  2021.0     3.0      0.0   
9370        9370             West Side Story  2021.0     3.0      0.0   
9371        9371     Every Time a Bell Rings  2021.0     3.0      0.0   
9372        9372          Single All The Way  2021.0     3.0      0.0   

          imdbid       id  submitterid                 date  visible  
0      3155794.0   9602.0      18880.0  2021-04-02 20:58:09      1.0  
1     14495706.0   9804.0      19145.0  2021-05-11 00:11:22      1.0  
2      2221420.0   9603.0      18882.0  2021-04-03 02:25:27      1.0  
3     12592084.0   9806.0      19151.0  2021-05-11 23:38:54      1.0  
4      7816420.0   9816.0      19162.0  2021-05-13 01:32:14      1.0  
...          ...      ...          ...                  ...      ...  
9368  10752004.0  10152.0      19735.0  2021-12-05 19:22:20      1.0  
9369   3228774.0   9861.0      19231.0  2021-06-01 03:16:58      1.0  
9370   3581652.0  10157.0      19743.0  2021-12-10 03:10:09      1.0  
9371  15943414.0  10158.0      19744.0  2021-12-10 08:03:02      1.0  
9372  14315756.0  10161.0      19753.0  2021-12-15 10:10:36      1.0  

[9373 rows x 10 columns]

In [55]:
df_old.drop(['Unnamed: 0'], axis=1,inplace=True)

I need to add films between index 9373 and 10135

## Get movie by imdbid

Try with one film request

In [3]:
pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(14315756), typ='series')

C:\Users\lucia\AppData\Local\Temp\ipykernel_7632\2789128519.py:1: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(14315756), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_7632\2789128519.py:1: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(14315756), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_7632\2789128519.

imdbid                    14315756
title           Single All The Way
date           2021-12-15 10:10:36
year                          2021
submitterid                  19753
visible                          1
dubious                          0
rating                           3
id                           10161
dtype: object

Obtain info for the new film

In [4]:
c = 9372

In [5]:
count = 1
# create an empty dataframe
df_new = pd.DataFrame(columns=['title', 'year', 'rating', 'dubious', 'imdbid', 'id', 'submitterid', 'date', 'visible'])

# call from the latest to the earliest movie, in case the website crashes
for imdbid in df['imdbid'][-1:c:-1]:
    series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
    #print(series)
    df_new.loc[len(df_new.index)] = series
    ################## df_new = df_new.append(series, ignore_index=True)
    # every 10 movies, we save to a csv file, 
    # so that we wouldn't lose everything when there is an error
    if count % 10 == 0:
        df_new.to_csv('../data/Bechdel_updated.csv')
    count += 1
    
# reverse 
df_new = df_new.iloc[::-1].reset_index(drop=True)

df_new.to_csv('../data/Bechdel_updated.csv',index=False)

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_19052\3977060043.py:7: FutureWarning: The behavior of 'to_datetime' with 'unit' when parsing strings is deprecated. In a future version, strings will be parsed as datetime strings, matching the behavior without a 'unit'. To retain the old behavior, explicitly cast ints or floats to numeric type before calling to_datetime.
  series = pd.read_json('http://bechdeltest.com/api/v1/getMovieByImdbId?imdbid=' + str(imdbid), typ='series')
C:\Users\lucia\AppData\Local\Temp\ipykernel_

In [52]:
df_new.sample(5)

title    year  rating  dubious      imdbid  \
619                       Athena  2022.0     2.0      0.0  15445056.0   
393  America: The Motion Picture  2021.0     3.0      1.0   6733874.0   
536                    Uncharted  2022.0     2.0      1.0   1464335.0   
76                         HELEN  2019.0     1.0      0.0  10308528.0   
89            Vast of Night, The  2019.0     3.0      0.0   6803046.0   

          id  submitterid                 date  visible  
619  10535.0      20489.0  2022-10-14 14:13:54      1.0  
393   9924.0      19325.0  2021-07-02 21:22:53      1.0  
536  10226.0      19901.0  2022-02-14 13:05:22      1.0  
76    9089.0      17887.0  2020-04-22 22:02:01      1.0  
89    9159.0      18064.0  2020-06-07 01:22:07      1.0

#### Concatenate the two datasets

In [57]:
df_new = pd.read_csv("../data/Bechdel_updated.csv", index_col=0)

In [58]:
df_new

Unnamed: 0.1  Unnamed: 0                             title    year  \
0               0         764                        Gemini Man  2019.0   
1               1         763                    Ford v Ferrari  2019.0   
2               2         762                          Hustlers  2019.0   
3               3         761      Maleficent: Mistress of Evil  2019.0   
4               4         760                   Laundromat, The  2019.0   
..            ...         ...                               ...     ...   
760           760           4  Winnie-the-Pooh: Blood and Honey  2023.0   
761           761           3                   Haunted Mansion  2023.0   
762           762           2                Portable Door, The  2023.0   
763           763           1          Anatomie d&#39;une chute  2023.0   
764           764           0                      Slotherhouse  2023.0   

     rating  dubious      imdbid       id  submitterid                 date  \
0       1.0      0.0   1025100.0   8868.0      17383.0  2019-10-12 10:46:05   
1       0.0      0.0   1950186.0   8869.0      17384.0  2019-10-12 10:48:31   
2       3.0      0.0   5503686.0   8870.0      17389.0  2019-10-14 13:49:26   
3       3.0      0.0   4777008.0   8876.0      17403.0  2019-10-26 22:39:21   
4       3.0      0.0   5865326.0   8877.0      17405.0  2019-10-27 07:11:37   
..      ...      ...         ...      ...          ...                  ...   
760     3.0      0.0  19623240.0  10995.0      21441.0  2023-09-03 11:21:01   
761     3.0      0.0   1695843.0  10996.0      21442.0  2023-09-03 18:32:34   
762     3.0      0.0  11820950.0  11009.0      21466.0  2023-09-16 03:43:53   
763     3.0      0.0  17009710.0  11010.0      21469.0  2023-09-19 08:14:23   
764     3.0      0.0  15422224.0  11015.0      21477.0  2023-09-24 01:56:22   

     visible  
0        1.0  
1        1.0  
2        1.0  
3        1.0  
4        1.0  
..       ...  
760      1.0  
761      1.0  
762      1.0  
763      1.0  
764      1.0  

[765 rows x 11 columns]

In [60]:
df_new.drop(['Unnamed: 0','Unnamed: 0.1'], axis=1,inplace=True)

In [61]:
df_new

title    year  rating  dubious      imdbid  \
0                          Gemini Man  2019.0     1.0      0.0   1025100.0   
1                      Ford v Ferrari  2019.0     0.0      0.0   1950186.0   
2                            Hustlers  2019.0     3.0      0.0   5503686.0   
3        Maleficent: Mistress of Evil  2019.0     3.0      0.0   4777008.0   
4                     Laundromat, The  2019.0     3.0      0.0   5865326.0   
..                                ...     ...     ...      ...         ...   
760  Winnie-the-Pooh: Blood and Honey  2023.0     3.0      0.0  19623240.0   
761                   Haunted Mansion  2023.0     3.0      0.0   1695843.0   
762                Portable Door, The  2023.0     3.0      0.0  11820950.0   
763          Anatomie d&#39;une chute  2023.0     3.0      0.0  17009710.0   
764                      Slotherhouse  2023.0     3.0      0.0  15422224.0   

          id  submitterid                 date  visible  
0     8868.0      17383.0  2019-10-12 10:46:05      1.0  
1     8869.0      17384.0  2019-10-12 10:48:31      1.0  
2     8870.0      17389.0  2019-10-14 13:49:26      1.0  
3     8876.0      17403.0  2019-10-26 22:39:21      1.0  
4     8877.0      17405.0  2019-10-27 07:11:37      1.0  
..       ...          ...                  ...      ...  
760  10995.0      21441.0  2023-09-03 11:21:01      1.0  
761  10996.0      21442.0  2023-09-03 18:32:34      1.0  
762  11009.0      21466.0  2023-09-16 03:43:53      1.0  
763  11010.0      21469.0  2023-09-19 08:14:23      1.0  
764  11015.0      21477.0  2023-09-24 01:56:22      1.0  

[765 rows x 9 columns]

In [62]:
bechdel_df = pd.concat([df_old,df_new],axis=0)

In [63]:
bechdel_df

title    year  rating  dubious      imdbid  \
0                    Passage de Venus  1874.0     0.0      0.0   3155794.0   
1                   La Rosace Magique  1877.0     0.0      0.0  14495706.0   
2          Sallie Gardner at a Gallop  1878.0     0.0      0.0   2221420.0   
3                   Le singe musicien  1878.0     0.0      0.0  12592084.0   
4             Athlete Swinging a Pick  1881.0     0.0      0.0   7816420.0   
..                                ...     ...     ...      ...         ...   
760  Winnie-the-Pooh: Blood and Honey  2023.0     3.0      0.0  19623240.0   
761                   Haunted Mansion  2023.0     3.0      0.0   1695843.0   
762                Portable Door, The  2023.0     3.0      0.0  11820950.0   
763          Anatomie d&#39;une chute  2023.0     3.0      0.0  17009710.0   
764                      Slotherhouse  2023.0     3.0      0.0  15422224.0   

          id  submitterid                 date  visible  
0     9602.0      18880.0  2021-04-02 20:58:09      1.0  
1     9804.0      19145.0  2021-05-11 00:11:22      1.0  
2     9603.0      18882.0  2021-04-03 02:25:27      1.0  
3     9806.0      19151.0  2021-05-11 23:38:54      1.0  
4     9816.0      19162.0  2021-05-13 01:32:14      1.0  
..       ...          ...                  ...      ...  
760  10995.0      21441.0  2023-09-03 11:21:01      1.0  
761  10996.0      21442.0  2023-09-03 18:32:34      1.0  
762  11009.0      21466.0  2023-09-16 03:43:53      1.0  
763  11010.0      21469.0  2023-09-19 08:14:23      1.0  
764  11015.0      21477.0  2023-09-24 01:56:22      1.0  

[10138 rows x 9 columns]

In [66]:
bechdel_df.reset_index(inplace=True)

In [67]:
bechdel_df

index                             title    year  rating  dubious  \
0          0                  Passage de Venus  1874.0     0.0      0.0   
1          1                 La Rosace Magique  1877.0     0.0      0.0   
2          2        Sallie Gardner at a Gallop  1878.0     0.0      0.0   
3          3                 Le singe musicien  1878.0     0.0      0.0   
4          4           Athlete Swinging a Pick  1881.0     0.0      0.0   
...      ...                               ...     ...     ...      ...   
10133    760  Winnie-the-Pooh: Blood and Honey  2023.0     3.0      0.0   
10134    761                   Haunted Mansion  2023.0     3.0      0.0   
10135    762                Portable Door, The  2023.0     3.0      0.0   
10136    763          Anatomie d&#39;une chute  2023.0     3.0      0.0   
10137    764                      Slotherhouse  2023.0     3.0      0.0   

           imdbid       id  submitterid                 date  visible  
0       3155794.0   9602.0      18880.0  2021-04-02 20:58:09      1.0  
1      14495706.0   9804.0      19145.0  2021-05-11 00:11:22      1.0  
2       2221420.0   9603.0      18882.0  2021-04-03 02:25:27      1.0  
3      12592084.0   9806.0      19151.0  2021-05-11 23:38:54      1.0  
4       7816420.0   9816.0      19162.0  2021-05-13 01:32:14      1.0  
...           ...      ...          ...                  ...      ...  
10133  19623240.0  10995.0      21441.0  2023-09-03 11:21:01      1.0  
10134   1695843.0  10996.0      21442.0  2023-09-03 18:32:34      1.0  
10135  11820950.0  11009.0      21466.0  2023-09-16 03:43:53      1.0  
10136  17009710.0  11010.0      21469.0  2023-09-19 08:14:23      1.0  
10137  15422224.0  11015.0      21477.0  2023-09-24 01:56:22      1.0  

[10138 rows x 10 columns]

In [69]:
bechdel_df.duplicated().value_counts()

False    10138
Name: count, dtype: int64

In [70]:
bechdel_df.to_csv('../data/Bechdel.csv',index=False)